prov/rxm/src/rxm_ep.c

```diff
@@ -168,8 +168,22 @@ static int rxm_recv_queue_init(struct rxm_recv_queue *recv_queue, size_t size,
 
 static void rxm_send_queue_close(struct rxm_send_queue *send_queue)
 {
-	if (send_queue->fs)
+	if (send_queue->fs) {
+		struct rxm_tx_entry *tx_entry;
+		ssize_t i;
+
+		fastlock_acquire(&send_queue->lock);
+		for (i = send_queue->fs->size - 1; i >= 0; i--) {
+			tx_entry = &send_queue->fs->buf[i];
+			if (tx_entry->tx_buf) {
+				rxm_buf_release(&tx_entry->ep->tx_pool,
+						(struct rxm_buf *)tx_entry->tx_buf);
+				tx_entry->tx_buf = NULL;
+			}
+		}
 		rxm_txe_fs_free(send_queue->fs);
+		fastlock_release(&send_queue->lock);
+	}
 	fastlock_destroy(&send_queue->lock);
```

Comment:
We destroy the lock here, so there's no need to acquire it above.  We can't be in a race condition, or the other thread attempting to acquire the lock is going to end up accessed one that is destroyed.

Reply:
Ok. Will remove acquiring this lock.
I've added it because Coverity can complain about it. Let's see what Coverity will say about this :)

Reply:
Ah - that makes sense.  I can mark it as a false positive once coverity complains again.

Reply:
> I can mark it as a false positive once coverity complains again.

Thanks 👍